src/ray/raylet/worker_pool.cc

```diff
@@ -146,6 +146,11 @@ Process WorkerPool::StartWorkerProcess(
   rpc::JobConfig *job_config = nullptr;
   if (!IsIOWorkerType(worker_type)) {
     RAY_CHECK(!job_id.IsNil());
+    if (finished_jobs_.contains(job_id)) {
```

Comment:
Why is check is needed in this PR?

Reply:
I removed the worker pool modifications from this PR.